zerver/lib/bugdown/__init__.py

```diff
@@ -698,6 +701,28 @@ def handleMatch(self, match):
         img.set('alt', email)
         return img
 
+class Timestamp(markdown.inlinepatterns.Pattern):
+    def handleMatch(self, match):
+        # type: (Match[Text]) -> Optional[Element]
+        span = markdown.util.etree.Element('span')
+        span.set('class', 'timestamp')
+        anchor = markdown.util.etree.SubElement(span, 'span')
+        try:
+            timestamp = dateutil.parser.parse(match.group('time'))
+            span.set('value', timestamp.strftime("%s"))
+            anchor.set('value', timestamp.strftime("%s"))
+        except ValueError:
+            try:
+                timestamp = datetime.fromtimestamp(float(match.group('time')))
+                span.set('value', timestamp.strftime("%s"))
+                anchor.set('value', timestamp.strftime("%s"))
+            except ValueError:
+                pass
```

Comment:
I would move the duplicate `span.set` lines outside of the try/except stuff; this code is harder to read than it should be.

Reply:
We could also check which possibility to try more simply by seeing if the string is an integer, rather than this try/except series.